samples/javascript_nodejs/02.echobot-with-counter/index.js

```diff
@@ -38,19 +27,38 @@ try {
 } catch (err) {
     console.error(`\nError reading bot file. Please ensure you have valid botFilePath and botFileSecret set for your environment.`);
     console.error(`\n - The botFileSecret is available under appsettings for your Azure Bot Service bot.`);
-    console.error(`\n - If you are running this bot locally, consider adding a .env file with botFilePath and botFileSecret.\n\n`);
+    console.error(`\n - If you are running this bot locally, consider adding a .env file with botFilePath and botFileSecret.`);
+    console.error(`\n - See https://aka.ms/about-bot-adapter to learn more about .bot file its use and bot configuration.\n\n`);
     process.exit();
 }
 
+// For local development configuration as defined in .bot file
+const DEV_ENVIRONMENT = 'development';
+
+// Bot name as defined in .bot file or from runtime
+const BOT_CONFIGURATION = (process.env.NODE_ENV || DEV_ENVIRONMENT);
+
 // Get bot endpoint configuration by service name
+// Bot configuration as defined in .bot file 
 const endpointConfig = botConfig.findServiceByNameOrId(BOT_CONFIGURATION);
 
-// Create adapter. See https://aka.ms/about-bot-adapter to learn more about .bot file its use and bot configuration .
+// Create bot adapter. 
+// See https://aka.ms/about-bot-adapter to learn more about bot adapter.
 const adapter = new BotFrameworkAdapter({
     appId: endpointConfig.appId || process.env.microsoftAppID,
     appPassword: endpointConfig.appPassword || process.env.microsoftAppPassword
 });
 
+// Catch-all for any unhandled errors in your bot.
+adapter.onTurnError = async (context, error) => {
+    // This check writes out errors to console log .vs. app insights.
+    console.error(`\n [onTurnError]: ${ error }`);
+    // Send a message to the user
+    context.sendActivity(`Oops. Something went wrong!`);
+    // Clear out state
+    conversationState.clear(context);
```

Comment:
Hmm. I'm not doing this.  Do we need to do this?

Reply:
@daveta  yes, we should add the catch-all exception